### ISSUE TITLE
OpenGL ES support ?

### ISSUE THREAD

**Issue by procfxgen** [MonoGame/MonoGame#5327]
Will the new SDL2 monogame platform support OpenglES ?

**Comment by tomspilman** [comment]
I believe it does.  @cra0zy @dellis1972 ?

**Comment by cra0zy** [comment]
I don't think it does, our previous code did not support GLES and all the new code did was replace OpenTK. Providing GLES support is something that would be nice, but would probably just end up being a waste of time (I can't think of a Linux device besides Raspberry Pi that would benefit from it).

**Comment by amerkoleci** [comment]
Why not? I can use Google angle and try gles 2 or 3 device on Windows to see how it would work on mobile devices

**Comment by KonajuGames** [comment]
> Why not? I can use Google angle and try gles 2 or 3 device on Windows to see how it would work on mobile devices

We have had ANGLE support for two or so years now, and it has had very little use.  I can't recall the last time it was mentioned in an issue or forum post.

**Comment by procfxgen** [comment]
Nvidia had opengles support on windows, to try it I used :
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_ES);
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 2);
	SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 0);

On monogame 3.6, android will used opentk not new sdl2platform ?

**Comment by mrhelmut** [comment]
> On monogame 3.6, android will used opentk not new sdl2platform ?
Android uses OpenTK, but a move to SDL may happen, which I believe would be the only chance for an ES support to be merged back into desktop platforms.

I really don't see any use of ES on desktop beside RaspPI, and they now have a full OpenGL support anyway (and are not powerful enough to make any relevant use of a Mono + MonoGame apps, as far as I have tested performances are terrible).

**Comment by dellis1972** [comment]
Android will NOT be using SDL2. It will be using the custom OpenGL bindings
though. There is a work in progress branch [1] which replaces OpenTK on iOS
, Android and MacOS. But it still needs work, and at the moment I am not
able to finish it off. If someone wants to work on it, let me know and I
will grant you access to my fork.

[1] https://github.com/dellis1972/MonoGame/commits/opentkandroid

**Comment by procfxgen** [comment]
I have done some tests with monogame 3.6 and experimental opengl support on raspbian.
maybe it's could be better with opengles ....

https://twitter.com/procfxgen/status/787387345033170946

**Comment by procfxgen** [comment]
I have done a sample that display a 3d model on a raspberry with opengles and SDL2.0.5 but without monogame framework.

That use a modified version of monogame SDL.cs and OpenGL.cs files.

So I have tried to integrate modified version to monogame framework without success ... that fail on drawprimites ...

My sample app...
[ConsoleSDL.zip](https://github.com/MonoGame/MonoGame/files/645891/ConsoleSDL.zip)

**Comment by procfxgen** [comment]
I have a patched monogame version that work on raspberry with SDL2.0.5 and custom openGL.
There are some works to do still with RenderTarget and Textures but it's a good start.
If you are interested I could clean up my code and send a patch...

**Comment by CartBlanche** [comment]
@KonajuGames , @fxgenstudio @amerkoleci I've just raised a PR to add OpenGL ES 3.x here - https://github.com/MonoGame/MonoGame/pull/9137, which so far works on iOS and Android.

**Comment by dellis1972** [comment]
@CartBlanche  that is not what there are asking. They want the SDL (aka DesktopGL) to support GLES

**Comment by RobertoFreireDev** [comment]
Hey, I have this MonoGame.Framework.DesktopGL project https://github.com/RobertoFreireDev/Game-Engine

It works fine on: 
- Windows 
- Raspbery Pi 5 using Pi OS Lite (No desktop environment) by adding 2 extra packages libsdl2-dev and rtkit

But I can't make work on my RG353PS Linux Handheld device.

I am not expert on this, but I think that the root issue is because I am trying to run a desktop OpenGL (OpenGL) on hardware that only supports **OpenGL ES**.

Error:

sudo ./blackbox.sh
Unhandled exception. Microsoft.Xna.Framework.Graphics.NoSuitableGraphicsDeviceException: Failed to create graphics device!
 ---> System.PlatformNotSupportedException: MonoGame requires either ARB_framebuffer_object or EXT_framebuffer_object.Try updating your graphics drivers.
   at Microsoft.Xna.Framework.Graphics.GraphicsDevice.FramebufferHelper.Create(GraphicsDevice gd)...

**Comment by tomspilman** [comment]
@CartBlanche @dellis1972 @ThomasFOG 

The new GL native support (see #9116) be evolved to also support GLES?

**Comment by ThomasFOG** [comment]
I really dunno if an ES support is still worth anything, most hardware that might benefit from it actually supports Vulkan. It's the case of the Anbernic stuff, the RG353PS supports Vulkan (and the Pi 4+ does too).

Trying to build the native backend for ```linux-arm64``` might be a better, more future proof effort than still focusing on GL.

**Comment by AndreMohren-SII** [comment]
Just joining in to watch for this. I also have an R36S which i have successfully run my 3d render engine prototype.

<img width="852" height="444" alt="Image" src="https://github.com/user-attachments/assets/b2cb8857-bba2-4fde-af1b-ce621c5eab06" />

However using GL4ES realy slows monogame experiments down. Just tried the latest preview build of vulkan, just to find out that its not provided for arm64. Would love to test this as soon as its available. GLES3 would be awesome, but an official vulkan arm64 too. It might be interesting to check some experiments with both if GLES3 is worth keeping due to being faster then the Vulkan implementation on those devices, or just completely redundant if not.

**Comment by ThomasFOG** [comment]
I would expect a Vulkan driver to be faster on such devices. Not because Vulkan is faster on paper, but because OpenGL drivers tend to be pretty bad and unmaintained these days, especially on ARM devices.
